openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,321 @@
+package managementcpusoverride
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	PluginName                              = "autoscaling.openshift.io/ManagementCPUsOverride"
+	podWorkloadAnnotation                   = "workload.openshift.io"
+	podWorkloadAnnotationEffect             = "effect"
+	workloadEffectPreferredDuringScheduling = "PreferredDuringScheduling"
+	managedCapacityLabel                    = "management.workload.openshift.io/cores"
+	containerResourcesAnnotationPrefix      = "io.openshift.workload"
+	containerCPUSharesAnnotation            = "io.openshift.workload.management.cpushares"
+	containerCPUSetAnnotation               = "io.openshift.workload.management.cpuset"
+	namespaceAllowedAnnotation              = "workload.openshift.io/allowed"
+	workloadTypeManagement                  = "management"
+)
+
+var _ = initializer.WantsExternalKubeInformerFactory(&managementCPUsOverride{})
+var _ = admission.MutationInterface(&managementCPUsOverride{})
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(config io.Reader) (admission.Interface, error) {
+			return &managementCPUsOverride{
+				Handler: admission.NewHandler(admission.Create),
+			}, nil
+		})
+}
+
+type managementCPUsOverride struct {
```

Comment:
describe what it does in one or two paragraphs.

Reply:
done

Reply:
This needs definitely more. I should understand the user-visible semantics by reading this.

Reply:
I found it in doc.go. Please add "See doc.go for more details" or move the docs to here. doc.go is not used much in our code base.

Reply:
I will copy-paste it from the doc